tests/tests_integration/test_api/test_assets.py

```diff
@@ -99,3 +99,16 @@ def test_post_asset_hierarchy(self, new_asset_hierarchy):
     def test_get_subtree(self, root_test_asset):
         assert 781 == len(COGNITE_CLIENT.assets.retrieve_subtree(root_test_asset.id))
         assert 6 == len(COGNITE_CLIENT.assets.retrieve_subtree(root_test_asset.id, depth=1))
+
+    def test_create_asset_hierarchy_parent_external_id_not_in_request(self):
+        root_external_id = "my_root_{}".format(utils._auxiliary.random_string(5))
+        root = Asset(external_id=root_external_id, name="my_root")
+        root = COGNITE_CLIENT.assets.create(root)
+
+        children = generate_asset_tree(
+            root_external_id=root_external_id, depth=5, children_per_node=10, current_depth=2
+        )
+        children = COGNITE_CLIENT.assets.create(children)
+        assert COGNITE_CLIENT.assets._CREATE_LIMIT < len(children)
```

Comment:
Why do make this assertion?

I think we should also test that the resulting hierarchy has the intended structure.

Reply:
It's just to make sure that the _AssetPoster is tested. I agree that we should test the structure as well.